components/experimental/src/duration/provider/digital.rs

```diff
@@ -0,0 +1,139 @@
+#[derive(Debug, Copy, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::duration::provider)
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+/// A struct containing digital duration data (durationUnit-type-* patterns).
+pub struct DigitalDurationDataV1 {
+    /// To represent the patterns from: durationUnit-type-hm
+    pub hm: HmVariant,
+
+    /// To represent the patterns from: durationUnit-type-hms
+    pub hms: HmsVariant,
+
+    /// To represent the patterns from: durationUnit-type-ms
+    pub ms: MsVariant,
+}
+
+#[zerovec::make_ule(HmVariantULE)]
+#[zerovec::derive(Debug)]
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::duration::provider::digital),
+)]
+#[repr(u8)]
+/// Represents possible variants of the hour-minute pattern.
+pub enum HmVariant {
+    /// hh:mm
+    DoubleHourColonSep = 0,
+    /// h:mm
+    SingleHourColonSep = 1,
+    /// h.mm
+    SingleHourDotSep = 2,
+}
+
+#[zerovec::make_ule(HmsVariantULE)]
+#[zerovec::derive(Debug)]
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::duration::provider::digital)
+)]
+#[repr(u8)]
+/// Represents possible variants of the hour-minute-second pattern.
+pub enum HmsVariant {
+    /// hh:mm:ss
+    DoubleHourColonSep = 0,
+    /// h:mm:ss
+    SingleHourColonSep = 1,
+    /// h.mm.ss
+    SingleHourDotSep = 2,
+}
+
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::duration::provider::digital)
+)]
+#[zerovec::make_ule(MsVariantULE)]
+#[zerovec::derive(Debug)]
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[repr(u8)]
+/// Represents possible variants of the minute-second pattern.
+pub enum MsVariant {
+    /// mm:ss
+    DoubleMinuteColonSep = 0,
+    /// m:ss
+    SingleMinuteColonSep = 1,
+    /// m.ss
+    SingleMinuteDotSep = 2,
+}
+
+use alloc::str::FromStr;
+use alloc::string::{String, ToString};
+
+/// Unknown time pattern: {0}
+#[derive(Debug, Display)]
+pub struct UnknownPatternError(String);
+
+impl FromStr for HmsVariant {
+    type Err = UnknownPatternError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hh:mm:ss" => Ok(HmsVariant::DoubleHourColonSep),
+            "h:mm:ss" => Ok(HmsVariant::SingleHourColonSep),
+            "h.mm.ss" => Ok(HmsVariant::SingleHourDotSep),
+            _ => Err(UnknownPatternError(s.to_string())),
+        }
+    }
+}
```

Comment:
You can implement `FromStr`  for  `DigitalDurationDataV1`

example:

```rust
 match s {
            "hh:mm:ss" => Ok(DigitalDurationDataV1{separator: ":", padding: 2}),
            ... etc.
        }
```

also, I think there is a pattern --> `"h,mm,ss"`